src/dsl/api-journey.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2020-present, Elastic NV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import { APIRequestContext } from 'playwright-core';
+import { Step } from './step';
+import {
+  VoidCallback,
+  Params,
+  Location,
+  StatusValue,
+  APIHooksCallback,
+} from '../common_types';
+import { Monitor, MonitorConfig } from './monitor';
+import { isMatch } from '../helpers';
+import { APIRunnerInfo } from '../core/api-runner';
+
+export type APIJourneyOptions = {
+  name: string;
+  id?: string;
+  tags?: string[];
+};
+
+type HookType = 'before' | 'after';
+export type APIHooks = Record<HookType, Array<APIHooksCallback>>;
+type APIJourneyCallbackOpts = {
+  params: Params;
+  request: APIRequestContext;
+  info: APIRunnerInfo;
+};
+export type APIJourneyCallback = (options: APIJourneyCallbackOpts) => void;
+
+export class APIJourney {
```

Comment:
Ideally I would like us to have a base Journey class that both BrowserJourney and APIJourney extends from and the runner should be a single entity that decides how to run each of them. I am not in favor of having multiple runners.